lib/chai/core/assertions.js

```diff
@@ -537,8 +537,8 @@ module.exports = function (chai, _) {
         itemsCount = val.length;
         break;
       case 'function':
-        var name = val.name ? ' ' + val.name : '';
-        throw new TypeError('.empty was passed a function' + name);
+        var msg = '.empty was passed a function ' + _.getName(val);
+        throw new TypeError(msg.trim());
```

Comment:
Ehhh, my feeling is that if someone goes out of there way to explicitly add spaces in such a weird way, then `getName` respects it and returns it as-is.